### ISSUE TITLE
Case sensitivity in OBS for bound names

### ISSUE THREAD

**Issue by mjr-deltares** [MODFLOW-USGS/modflow6#1131]
**Describe the bug**
Error when referring to bound name from observations file in lower case

**To Reproduce**
Define a GWFGWF exchange with boundname:

```
BEGIN exchangedata
  1 1 5  1 1 1  1     50.00000000      50.00000000     100.00000000    lowercase_name
END exchangedata
```
and add an observation in the exchange's OBS file referring to the former as

```
BEGIN continuous  FILEOUT  gwf_obs_names.csv
  exg1_name  FLOW-JA-FACE  lowercase_name
END continuous  FILEOUT  gwf_obs_names.csv
```
this 'll give the message

```
ERROR REPORT:

  1. Exchange "name1" for observation "FLOW-JA-FACE" is invalid in package
     "GWF-GWF_1"
```

**Expected behavior**
Specifying both in lower case should probably work. If I change the OBS boundname to upper case, it works fine.

**Comment by langevin-usgs** [comment]
I think our convention for names is to convert them all to uppercase upon reading.  Users can provide any case, but all internal storage and output is upper case.  So this looks like something we need to fix.

**Comment by mjr-deltares** [comment]
Yes, agree, I will pick up the issue

**Comment by mjr-deltares** [comment]
this is fixed as of PR #1140
